### ISSUE TITLE
Menu item to update database & Calibre-Web fails: can no longer log in, "502 Bad Gateway"

### ISSUE THREAD

**Issue by holta** [iiab/calibre-web#31]
Additionally, There’s a menu to update database and update calibre,tried that and now can’t logon,bad gateway
![2B1E1FF8-2889-4BBA-9F19-DC339D12353F](https://github.com/iiab/calibre-web/assets/52469992/b1017d11-32e8-4d6d-8972-8d7c0d487f3a)

_Originally posted by @EMG70 in https://github.com/iiab/calibre-web/issues/29#issuecomment-1806235139_

**Comment by holta** [comment]
@EMG70 can you confirm the problem happens after you do... ?

Log in to **Calibre-Web > Admin** (on top, by Tasks button) **> Version Information > Check for Update > Perform Update**

![image](https://github.com/iiab/calibre-web/assets/2458907/9c87debb-ec4e-40e2-8974-04bbb6580d81)

(We definitely might need to hide the yellow sentence "Click on the button below to update to the latest stable version." in the interim, until this is later fixed and working!)

**Comment by EMG70** [comment]
Yes the menu for update came after clicking admin next to Task as shown in screenshots.

**Comment by EMG70** [comment]
![IMG_1754](https://github.com/iiab/calibre-web/assets/52469992/f9bd5558-fc32-4071-a86d-252c12b6f0c2)
![IMG_1756](https://github.com/iiab/calibre-web/assets/52469992/e736de73-2eb5-4296-940e-ef0a4ac6d149)

**Comment by holta** [comment]
Is a loud UX warning perhaps sufficient / best for now?

> DANGEROUS!

> SOFTWARE UPDATES ARE NOT SUPPORTED AS OF NOVEMBER 2023!

**Comment by holta** [comment]
ASIDE:

If absolutely necessary, we can add a "Known Issues" section to our ALPHA "Release Notes" for 2023-11-30 — cautioning people as to what's not yet working.

**Comment by holta** [comment]
This dangerous "Check for Update" needs to be either (a) blocked/removed or (b) fixed.

(As more people begin to get involved, the last thing we need is a shiny self-destruct button, that will in fact cause deep anger.)

**Comment by holta** [comment]
> About 2/3 of a year later, we need to move forward strongly considering fixing the "Check for Update" button soon — with broad community involvement taken much more seriously in 2024 than in 2023:

@deldesir I strongly recommend this be solved in August:

Allowing for much stronger (& more organic) community building henceforth!

**Comment by holta** [comment]
@deldesir just a reminder that Rock Solid Upgrading of xklb and yt-dlp must be a part of IIAB Calibre-Web's automated upgrade, — that is becoming much more urgent in 2024 — a year later now:

- The `pipx reinstall xklb` command is probably good enough, to upgrade core dependencies.
- But if absolutely necessary, we can consider [other approaches](https://github.com/iiab/calibre-web/wiki/Home/_compare/0c2d6be3cf6c62b9a6af25e49bdb5e028fdd43dd...38252fb54a8da1902651264a006e8694c926f73c).

**Comment by holta** [comment]
Just FYI there's a bit of progress that should help IIAB Calibre-Web provide an actual button to update itself — as soon as we can in coming weeks?

Thanks to the brand new CLI command [iiab-update](https://github.com/iiab/iiab/blob/master/scripts/iiab-update) that seems to work very well: (run `iiab-update -f` to upgrade Calibre-Web very quickly, in about 1 single minute!)

- PR iiab/iiab#3768
- PR iiab/iiab#3769
- PR iiab/iiab#3770
- PR iiab/iiab#3771

_Documentation below, to help non-technical folks start trying it out too:_

- https://wiki.iiab.io/go/FAQ#Can_I_upgrade_IIAB_software%3F
- https://github.com/iiab/iiab/tree/master/roles/calibre-web#upgrading
- https://github.com/iiab/calibre-web/wiki#upgrading

**Comment by holta** [comment]
> brand new CLI command [iiab-update](https://github.com/iiab/iiab/blob/master/scripts/iiab-update) [works well!}

> (run `iiab-update -f` to upgrade Calibre-Web very quickly, in about 1 single minute!)

Now beefed up with:

- PR iiab/iiab#3772
- PR iiab/iiab#3773
- PR iiab/iiab#3774
- PR iiab/iiab#3775
